### DISCUSSION TITLE
Whether Fast.farm can use FLORIS as controller?

### DISCUSSION THREAD

**Discussion by tatesnow** [OpenFAST/openfast#934] [Q&A] [answered]
Hi  @jjonkman ,
I am reading the 'FAST.Farm user's guide and theory manual', there‘s a description of the super controller 'The logic of such a super controller could be developed through the application of the National Renewable Energy Laboratory (NREL) code FLOw Redirection and Induction in Steady state (FLORIS)', I am not sure weather Fast.farm can use FLORIS as controller?

**Comment by jjonkman** [comment, marked as answer]
Dear @tatesnow,

FLORIS is a steady-state solver used to find optimal yaw angles of wind turbines in wind farms to maximize energy capture through wake steering. Typically you'd use FLORIS to derive a look-up table of yaw angles as a function of wind condition (wind direction, turbulence level / stability). But FLORIS will not generate the super controller source code or dynamic library for you. And you don't necessarily need to implement a super controller to implement different, but fixed, yaw angles for each wind turbine in the wind farm within FAST.Farm. Instead, you can implement these fixed yaw angles directly within the OpenFAST models and change the yaw angles when simulating different wind conditions in separate FAST.Farm simulations. 

Best regards,

**Comment by tatesnow** [comment]
Hi @jjonkman,
Thank you very much for your reply! Your answer perfectly answered my doubts! Do you have more information about the super controller of fast.farm? I really need it.

**Reply by jjonkman** [reply]
Hi @tatesnow,

The super controller of FAST.Farm is well documented in the online FAST.Farm User's Guide and Theory Manual--see: https://openfast.readthedocs.io/en/main/source/user/fast.farm/ModelGuidance.html#super-controller.

Best regards,
